### DISCUSSION TITLE
Release of version 2.20

### DISCUSSION THREAD

**Discussion by polluks** [cc65/cc65#1639] [Ideas]
Are there any critical open points?
How about deadline 2/20 😉

**Comment by mrdudz** [comment]
We are not in a hurry, are we? Right now I am in the process of cleaning some things up, and then i also want to produce proper github actions for making releases. And perhaps close/finish a few more PRs. It will still take a while for everything to settle :)

**Comment by mrdudz** [comment]
#1642 is a blocker

**Reply by polluks** [reply]
Greg saved your ass 😝

**Comment by WayneParham** [comment]
Is there a version 2.20 planned?

I guess that's a dumb question - since v.2.20 is the natural next after v.2.19 - but what I'm asking actually is what the team considers when making a new (major and minor) version rather than un-versioned bug fixes?

I assume we don't increment minor version numbers for each bugfix pull request, but rather allow the verson release date to identify those kinds of changes in packages.

So what is the style of versioning we use for cc65?

**Comment by polluks** [comment]
CI/CD is modern.
However we have the big roadmap https://github.com/cc65/wiki/wiki/Roadmap
If you break compatibility a new release number is always your friend...

2022-02-06 19:18 GMT+01:00, WayneParham ***@***.***>:

**Comment by mrdudz** [comment]
> So what is the style of versioning we use for cc65?

I don't understand that question :) I don't really care about numbering schemes myself, as long as it is increasing and a release marks a somewhat stable point in time. That said when someone asks me what version he should use, my answer is always "the head revision from git" - since we dont backport patches or bugfixes anyway, chances are the head revision works better than the last release.

> However we have the big roadmap

That "roadmap" is 99% wishful thinking though (and i doubt anything listed there except floating point support will ever happen, not even in the far future)

Anyway, as said above, there are still a couple things i want to solve before i even think about making another release. The best way to speed up the process is looking at bugs and fixing them :)

**Comment by WayneParham** [comment]
I understand and agree on the "bugfix revision" scheme of simply referring to a version by its git SHA hash.  But that's a source version and once a package is compiled to create a snapshot, I don't see any auto-versioning mechanism.  - Not that I'm championing for one - But I just wondered if we had a formal or informal way of defining major and minor release version names:
- https://github.com/cc65/cc65/releases

**Comment by mrdudz** [comment]
The compiled snapshots should also show the github hash in their --version output.

How the actual releases are being compiled and published is one of the things i still want to fix before doing actual releases (this should be all automatic when pushing a git tag for a new release). I don't know why there are no binaries linked for the older releases for that matter.

**Comment by WayneParham** [comment]
That's excellent, Bob.  I like the CI/CD approach - to automatically build and deploy a release candidate after every git merge/push.

One of the contract projects I'm working on has a manifest file that defines the major and minor version name of every build, which they then consider to be a release candidate.  If it passes all the regression tests and all the stakeholders approve the change request for a particular build, then it is released into production.  It is identified by the snapshot name, which is defined in the manifest file with major version being the two-digit year and minor version being the Julian date.  They do not actually auto-build on every push in this project - they kick off a daily build - so the snapshot naming convention never has a filename collision.

That approach works well creating a unique name for every production release build.  But I kind of like the way it has been done in cc65 too, at least in the past.  It appears that it wasn't automatic but I think Oliver Schmidt (and perhaps other team members) decided when there were enough incremental changes to warrant creating a new minor version.  I'm guessing major versions didn't shift except at times of major code shifts, like maybe when Ullrich von Bassewitz broke cc65 off from John Dunning's version of C, expanding it from a v.1.x to a v.2.x.  Totally guessing here, I don't know.  Which is why I was asking.  :-)

Anyway, enough of my questions and meandering.  Thanks so much for all you're doing.  I've really appreciated getting to visit with guys like you and Oliver, and the others cc65 contributors and team members here.   Super groovy cool work.

**Comment by mrdudz** [comment]
BTW, you can find the latest snapshot compile here: https://sourceforge.net/projects/cc65/files/

We can make something that keeps the last N versions prebuild on github, however we will have to produce git tags for this in some way (required by github), which i dont like much (tagging every commit basically defeats the purpose of tags). We'll see :)
